parlai/crowdsourcing/tasks/qa_data_collection/demo_worlds.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+from parlai.mturk.core.worlds import MTurkOnboardWorld, MTurkTaskWorld
+from parlai.core.worlds import validate
+import mephisto.data_model.dataloader as DataLoader
+
+
+class QADataCollectionOnboardWorld(MTurkOnboardWorld):
+    def __init__(self, opt, mturk_agent):
+        super().__init__(opt, mturk_agent)
+        self.opt = opt
+
+    def parley(self):
+        self.mturk_agent.agent_id = "Onboarding Agent"
+        self.mturk_agent.observe({"id": "System", "text": "Welcome onboard!"})
+        x = self.mturk_agent.act(timeout=self.opt["turn_timeout"])
+        self.mturk_agent.observe(
+            {
+                "id": "System",
+                "text": "Thank you for your input! Please wait while "
+                "we match you with another worker...",
+                "episode_done": True,
+            }
+        )
+        self.episodeDone = True
+
+
+class QADataCollectionWorld(MTurkTaskWorld):
+    """
+    World for recording a turker's question and answer given a context.
+    Assumes the context is a random context from a given task, e.g. from SQuAD, CBT,
+    etc.
+    """
+
+    collector_agent_id = 'QA Collector'
+
+    def __init__(self, opt, agent):
+        self.dataloader = opt["dataloader"]
+        self.agent = agent
+        self.agent.agent_id = "QA Agent"
+        self.episodeDone = False
+        self.turn_index = -1
+        self.context = None
+        self.question = None
+        self.answer = None
+        self.opt = opt
+
+    def parley(self):
+        # Each turn starts from the QA Collector agent
+        self.turn_index = (self.turn_index + 1) % 2
```

Comment:
Sorry, it got copied from another task, where we were having 3 turn conversation - question, answer, evidence

The current implementation asks question and answer in the same parley call i.e. parley is called only once. Can this lead to any unintended consequences? If so, I can add return statements [here](https://github.com/facebookresearch/ParlAI/pull/3318/files#diff-96eafa5491b76bc19d4d317fb7a6fccb9aeeb9a179b33a1b87a70b4ade3d5fb7R72)

Reply:
Hmm, I don't think there should be any unintended consequences - I just checked, and in the Mephisto repo it just looks like `world.parley()` is called over and over until `world.episode_done()` returns True. However, I just checked the documentation, and the docstring for `CrowdTaskWorld.parley()` states that a parley should represent one turn of your task. So I think the cleanest thing here would be to add a `return` statement at the end of the `if not self.question:` block, yes, so that `.parley()` will always be called exactly twice